ocictl/cmd/build.go

```diff
@@ -162,5 +118,53 @@ func (b *buildCmd) run(args []string) error {
 
 	}
 
-	return nil
+	builder := oci.Builder{
+		Logger: logger,
+		Client: cli,
+	}
+
+	res := make(chan v1alpha1.OCIBuildResponse)
+	errChan := make(chan error)
+	finished := make(chan bool)
+
+	go builder.Build(ociBuilderSpec, res, errChan, finished)
+
+	for {
+		select {
+
+		case err := <-errChan:
+			{
+				if err != nil {
+					logger.WithError(err).Errorln("error received from error channel whilst building")
+					return err
```

Comment:
Make sure to cancel or close the channels here - look at context with cancel

Reply:
context WithCancel is not appropriate in this case as we would not be able to run cancel in the caller function https://stackoverflow.com/a/46848395. We want to ideally call close in builder.go once it is finished and not within the ocictl cmd. Alternatively, i've added defers to close all channels once everything exits.